BUILD.bazel

```diff
@@ -0,0 +1,68 @@
+package(default_visibility = ["//visibility:private"])
+
+genrule(
+    name = "re2c_config",
+    srcs = [
+        "CMakeLists.txt",
+        "//bazel:re2c_config.h",
+    ],
+    outs = ["config.h"],
+    cmd = """
+        cat $(location //:CMakeLists.txt) |\\
+            sed -rne 's/project\\(re2c VERSION ([^ ]+).*/#define PACKAGE_VERSION "\\1" RE2C_VERSION_SUFFIX/p' >> "$@"
+        cat $(location //bazel:re2c_config.h) >> "$@"
+    """,
+)
+
+cc_library(
+    name = "re2c_config_cc",
+    hdrs = ["config.h"],
+    defines = ["""RE2C_STDLIB_DIR='"$(BINDIR)/re2c.runfiles/re2c/include"'"""],
+    data = ["include/unicode_categories.re"],
+)
+
+cc_library(
+    name = "re2c_base_lib_cc",
+    includes = ["bootstrap"],
+    hdrs = glob(["src/**/*.h"], exclude = [
+        "config.h",                    # Generated from 're2c_config_cc'
+        "src/parse/input.h",           # Depends on bootstrap sources
+        "src/util/nowarn_in_bison.h",  # Depends on bootstrap sources
+    ]),
+    srcs = glob(["src/**/*.cc"], exclude = [
+        "src/codegen/output.cc",       # PACKAGE_VERSON should be a function
+        "src/main.cc",                 # Binary only
+        "src/msg/msg.cc",              # PACKAGE_VERSON should be a function
+        "src/options/opt.cc",          # Depends on bootstrap sources
+        "src/options/symtab.cc",       # Depends on bootstrap sources
+        "src/parse/input.cc",          # Depends on bootstrap sources
+    ]),
+    deps = [
+        "re2c_config_cc",              # Needed for 're2c_config_cc' data files
+    ],
+)
+
+cc_library(
+    name = "re2c_bootstrap_cc",
```

Comment:
Still unresolved a question from previous patchset: why do you need to split the sources into a few libraries and then build the final binary from them? Now that you only use pre-generated bootstrap files, can't you just grab all the sources and build the executable in one go?